cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -941,6 +1019,8 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
                 return this.getPublishModal(xblockInfo, options);
             } else if (type === 'highlights') {
                 return this.getHighlightsModal(xblockInfo, options);
+            } else if (type === 'highlights_enable') {
+                return this.getHighlightsEnableModal(xblockInfo, options);
             } else {
```

Comment:
Do we need this else? Will it ever return null? If not, why not make change the "else if" above this the "else" condition?

Reply:
I believe we need this `else` in order to prevent accidental issues (while developing) where we return the incorrect modal when a new `type` is specified.

Reply:
It's possible this function could be later called with a garbage string that doesn't match any of these. In JavaScript, functions return `undefined` by default, so I'm guessing this is here so callers understand that they requested a modal that doesn't exist and not that the function just isn't returning anything. I don't think we would want to default to one particular modal in that case.